crates/primitives/src/receipt.rs

```diff
@@ -204,16 +204,17 @@ impl<'a> arbitrary::Arbitrary<'a> for Receipt {
     }
 }
 
-impl ReceiptWithBloom {
+/// A trait for enveloped encoding of receipts.
+pub trait EnvelopedEncoding {
     /// Returns the enveloped encoded receipt.
-    ///
-    /// See also [`ReceiptWithBloom::encode_enveloped`]
-    pub fn envelope_encoded(&self) -> Bytes {
-        let mut buf = Vec::new();
-        self.encode_enveloped(&mut buf);
-        buf.into()
-    }
+    fn envelope_encoded(&self) -> Bytes;
+
+    /// Encodes the receipt into its "raw" format.
+    fn encode_enveloped(&self, out: &mut dyn bytes::BufMut);
+}
```

Comment:
we don't need to introduce this trait,

instead this should implement the existing trait like:

https://github.com/paradigmxyz/reth/blob/5aceb3e11ed97be89a2cea338f904fc0afbc6dd4/crates/primitives/src/transaction/mod.rs#L1522-L1522